File: dotnet/ThirtyFour.UI.Tests/Window/TitleMatcherTests.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using ThirtyFour.UI.Window;

namespace ThirtyFour.UI.Tests.Window
{
    [TestClass]
    public class TitleMatcherTests : BaseTestSuite
    {

        [TestMethod]
        public void ExceptionThrownWhenNoMatchingTitle()
        {
            try
            {
                new TitleMatcher("shjksjd").MatchWindow(driver, 4);
                Assert.Fail("An exception should have been thrown");
            }
            catch (Exception e)
            {
                Assert.AreEqual("Unable to find matching window", e.Message);
            }
        }

        [TestMethod]
        public void WindowSwitchedWhenMatchingTitle()
        {

            driver.FindElement(By.LinkText("Popup")).Click();
            new TitleMatcher(@"GitHub - danhumphrey/ThirtyFour.UI: A Selenium (element 34) WebDriver UI Library for C# and Java").MatchWindow(driver, 4);
            Assert.AreEqual("https://github.com/danhumphrey/ThirtyFour.UI", driver.Url);
        }

        [TestMethod]
        public void WindowSwitchedWhenMatchingTitleOfDelayedWindow()
        {
            driver.FindElement(By.LinkText("Delayed Popup")).Click();
            new TitleMatcher(@"GitHub - danhumphrey/ThirtyFour.UI: A Selenium (element 34) WebDriver UI Library for C# and Java").MatchWindow(driver, 5);
            Assert.AreEqual("https://github.com/danhumphrey/ThirtyFour.UI", driver.Url);
        }
    }
}

File: java/src/test/java/com/github/danhumphrey/thirtyfour/ui/test/window/TitleMatcherTests.java
package com.github.danhumphrey.thirtyfour.ui.test.window;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

import com.github.danhumphrey.thirtyfour.ui.test.BaseTestSuite;
import com.github.danhumphrey.thirtyfour.ui.window.TitleMatcher;

public class TitleMatcherTests extends BaseTestSuite {

	@Test
	public void testExceptionThrownWhenNoMatchingTitle() {

		WebDriver driver = getDriver();
		String currentHandle = driver.getWindowHandle();

		try {
			new TitleMatcher("shjksjd").matchWindow(driver, 1000);
			Assertions.fail("An exception should have been thrown");
		} catch (Exception e) {
			Assertions.assertEquals("Unable to find a matching window", e.getMessage());
		} finally {
			quitExtraDrivers(currentHandle);
		}
	}

	@Test
    public void WindowSwitchedWhenMatchingTitle() throws Exception
    {
    	WebDriver driver = getDriver();
		String currentHandle = driver.getWindowHandle();

		driver.findElement(By.linkText("Popup")).click();
		
		try {
	        new TitleMatcher("GitHub - danhumphrey/ThirtyFour.UI: A Selenium (element 34) WebDriver UI Library for C# and Java").matchWindow(driver, 1000);
	        Assertions.assertEquals("https://github.com/danhumphrey/ThirtyFour.UI", driver.getCurrentUrl());
		} finally {
			quitExtraDrivers(currentHandle);
		}
        
    }

	@Test
    public void testWindowSwitchedWhenMatchingTitleOfDelayedWindow() throws Exception
    {
    	WebDriver driver = getDriver();
		String currentHandle = driver.getWindowHandle();
		
        driver.findElement(By.linkText("Delayed Popup")).click();
        
        try {
	        new TitleMatcher("GitHub - danhumphrey/ThirtyFour.UI: A Selenium (element 34) WebDriver UI Library for C# and Java").matchWindow(driver, 5000);
	        Assertions.assertEquals("https://github.com/danhumphrey/ThirtyFour.UI", driver.getCurrentUrl());
		} finally {
			quitExtraDrivers(currentHandle);
		}
    }
}
